src/errors/error-mapping.ts

```diff
@@ -0,0 +1,90 @@
+/*
+ * SPDX-FileCopyrightText: 2022 The HedgeDoc developers (see AUTHORS file)
+ *
+ * SPDX-License-Identifier: AGPL-3.0-only
+ */
+import {
+  ArgumentsHost,
+  BadRequestException,
+  ConflictException,
+  InternalServerErrorException,
+  NotFoundException,
+  UnauthorizedException,
+  UnprocessableEntityException,
+} from '@nestjs/common';
+import { HttpException } from '@nestjs/common/exceptions/http.exception';
+import { BaseExceptionFilter } from '@nestjs/core';
+
+import {
+  buildHttpExceptionObject,
+  HttpExceptionObject,
+} from './http-exception-object';
+
+type HttpExceptionConstructor = (object: HttpExceptionObject) => HttpException;
+
+const mapOfHedgeDocErrorsToHttpErrors: Map<string, HttpExceptionConstructor> =
+  new Map([
+    ['NotInDBError', (object): HttpException => new NotFoundException(object)],
+    [
+      'AlreadyInDBError',
+      (object): HttpException => new ConflictException(object),
+    ],
+    [
+      'ForbiddenIdError',
+      (object): HttpException => new ConflictException(object),
+    ],
+    ['ClientError', (object): HttpException => new BadRequestException(object)],
+    [
+      'PermissionError',
+      (object): HttpException => new UnauthorizedException(object),
+    ],
+    [
+      'TokenNotValidError',
+      (object): HttpException => new UnauthorizedException(object),
+    ],
+    [
+      'TooManyTokensError',
+      (object): HttpException => new BadRequestException(object),
+    ],
+    [
+      'PermissionsUpdateInconsistentError',
+      (object): HttpException => new UnprocessableEntityException(object),
```

Comment:
According to Wikipedia, HTTP 422 is WebDAV-specific

Reply:
Wikipedia is right. HTTP 422 is defined in [RFC 4918](https://datatracker.ietf.org/doc/html/rfc4918#section-11.2) which is the RFC defining WebDAV.

Reply:
Well that's not good then, should we replace it with `BadRequest` or something else?

Reply:
`BadRequest` will be used